### PR TITLE
Display MODS title and MODS alternative title if present in MODS record

### PR DESCRIPTION
**Pull request by BrandonMorr**
UWS has requested that we create a functionality that changes the object view page's title to mods:titleInfo/mods:title and mods:titleInfo[@ type="alternative"]/mods:title. The title only changes if either values are found in the MODS record.

### CODE REVIEW

File: uwsdora.module [hunk 1]
```diff
@@ -156,3 +156,31 @@ function uwsdora_convert_mods_to_citeproc_jsons_alter(&$output, $mods) {
...
+      $results = $xpath->query('mods:titleInfo[not (contains(@type, "alternative"))]/mods:title');
+      $node = $results->item(0);
+      $title = (isset($node)) ? $node->textContent : "";
+
+      $results = $xpath->query('mods:titleInfo[contains(@type, "alternative")]/mods:title');
+      $node = $results->item(0);
+      $title_alternative = (isset($node)) ? $node->textContent : "";
+
+      if (!empty($title) || !empty($title_alternative)) {
+        drupal_set_title(t("<span class='uwsdora-mods-title'>@title1</span><br/> <span class='uwsdora-mods-title-alternative'>@title2</span>", array('@title1' => $title, '@title2' => $title_alternative)), PASS_THROUGH);
```
Comment (jordandukart) [inline uwsdora.module]: If either `$title` or `$title_alternative` you have a `<span>` with nothing in it and in the first case a BR precluding the only title.

File: uwsdora.module [hunk 2]
```diff
@@ -156,3 +156,31 @@ function uwsdora_convert_mods_to_citeproc_jsons_alter(&$output, $mods) {
...
+
+    if ($document) {
+      $xpath = new DOMXpath($document);
+      $xpath->registerNamespace('mods', 'http://www.loc.gov/mods/v3');
+
+      $results = $xpath->query('mods:titleInfo[not (contains(@type, "alternative"))]/mods:title');
+      $node = $results->item(0);
+      $title = (isset($node)) ? $node->textContent : "";
+
+      $results = $xpath->query('mods:titleInfo[contains(@type, "alternative")]/mods:title');
```
Comment (jordandukart) [inline uwsdora.module]: These paths should either be full that is `/mods:mods/mods:titleInfo` or `//mods:titleInfo`.

File: uwsdora.module [hunk 3]
```diff
@@ -156,3 +156,31 @@ function uwsdora_convert_mods_to_citeproc_jsons_alter(&$output, $mods) {
...
+
+  if (isset($object['MODS'])) {
+    $document = new DOMDocument();
+    $document->loadXML($object['MODS']->content);
+
+    if ($document) {
+      $xpath = new DOMXpath($document);
+      $xpath->registerNamespace('mods', 'http://www.loc.gov/mods/v3');
+
+      $results = $xpath->query('mods:titleInfo[not (contains(@type, "alternative"))]/mods:title');
```
Comment (jordandukart) [inline uwsdora.module]: Whynot just `/mods:mods/mods:titleInfo[not(@type)]/mods:title`?

File: uwsdora.module [hunk 4]
```diff
@@ -156,3 +156,31 @@ function uwsdora_convert_mods_to_citeproc_jsons_alter(&$output, $mods) {
...
+      $node = $results->item(0);
+      $title = (isset($node)) ? $node->textContent : "";
```
Comment (jordandukart) [inline uwsdora.module]: You can simplify these queries into the following:

`$xpath->evaluate('normalize-space(//mods:titleInfo[not(@type)]/mods:title)');` and `$xpath->evaluate('normalize-space(//mods:titleInfo[@type="alternative"]/mods:title)');`

[Evaluate] (http://php.net/manual/en/domxpath.evaluate.php)for reference.

**Review by jordandukart — CHANGES_REQUESTED** [review]

File: uwsdora.module (hunk 1 above)
Comment (BrandonMorr) [reply uwsdora.module]: I've changed this to only render a new title if both values are present. Otherwise do nothing.

**Pull request merged**